pkg/user/controllers/user.go

```diff
@@ -83,23 +92,31 @@ func (uc *UserController) Create(w http.ResponseWriter, r *http.Request) {
 // @Router /tokens [post]
 // CreateToken handles the server request and calls CreateTokenInteractor
 func (uc *UserController) CreateToken(w http.ResponseWriter, r *http.Request) {
-	var user entities.User
-	err := json.NewDecoder(r.Body).Decode(&user)
+	var req CreateTokenRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
 	if err != nil {
 		uc.logger.Error("failed to parse request body")
 		uc.writeResponse(w, http.StatusUnprocessableEntity, nil)
 		return
 	}
 
-	token, err := uc.createTokenInteractor.Execute(user)
+	if req.Type == "" {
+		req.Type = "user"
+	}
+	if req.Duration == 0 {
+		req.Duration = 31536000
```

Comment:
Can't use `time.Duration` or `time.Hour` ?

Reply:
Additionally, if this change is applied, how will this impact the user input? For example, as is now, I'm able to set the duration in seconds simply by passing some `int` value... if we update, will the input format change? If yes, how so?

Reply:
The Duration type is a int64. See: https://golang.org/pkg/time/#Duration
We can use ParseDuration too if we think that the user can pass like 1s or 2h. See: https://golang.org/pkg/time/#ParseDuration

Reply:
Awesome :heart:

Reply:
I end up deciding to not use it for some reasons:
- the `Duration` type do not provide a more simple way to handle long slices of time such days and years
- the use of `Duration` type will make the code a little bit more complex, adding error checking while creating a `Duration` variable
- it also will need to add two additional operations: convert the seconds received from client request to `time.Duration`, and finally convert the `Duration` variable to seconds, to be able to send a mainflux authn proxy request

If you still think it'd better to use it instead of raw integers, please comment here! :smile: :+1:

Reply:
Seems good points to me. Resolving this comment